tests/robottelo/test_helpers.py

```diff
@@ -118,3 +119,15 @@ def test_slugify_component():
     assert slugify_component('File-Management', False) == 'file_management'
     assert slugify_component('File&Management') == 'filemanagement'
     assert slugify_component('File and Management') == 'filemanagement'
+
+
+class TestGetAvailableCapsulePort:
+    """Tests for method ``get_available_capsule_port``."""
+
+    @mock.patch('robottelo.helpers.ssh')
+    def test_return_port(self, ssh):
+        """get_available_capsule_port returns a port number.
+        """
+        ssh.command = mock.MagicMock(return_value=FakeSSHResult(['""'], 0))
```

Comment:
Can someone explain what's happening here? and how it's relevant to `get_available_capsule_port`?

Reply:
`ssh.command` function is replaced with mock, that will always return object specified in `return_value` when original function is called. This way `get_available_capsule_port` will not try to connect to SSH server, something that must fail on travis and is just too costly in regular unit test run.

If you are looking for general discussion about mocks (what they are, what purpose they serve, what are their downsides), you can give following two articles a try:
* [How to mock in Python? – (almost) definitive guide](https://breadcrumbscollector.tech/how-to-mock-in-python-almost-definitive-guide/)
* [Beware of chicken testing! (or mocks overuse)](https://breadcrumbscollector.tech/beware-of-chicken-testing-or-mocks-overuse/)

If you are interested in `MagickMock` class specifically, you can read [`unittest.mock`](https://docs.python.org/3/library/unittest.mock.html) documentation.